### ISSUE TITLE
Syntax highlighting

### ISSUE THREAD

**Issue by kamilkobak** [toolstack/Folio#30]
Currently, the best tool for keeping this type of notes is Obsidian, but it is not licensed under OpenSource. Folio may be an interesting replacement for Obsidian, but it still lacks a lot of functionality.

* First, Syntax highlighting.
* Automatic closing of blocks such as ``` or `
* Access to the data directory via WebDav

Good job

**Comment by toolstack** [comment]
Thanks, Folio will unlikely become a replacement for Obsidian.  While I don't use it, my understanding is that it is a very robust app, while Folio focuses on a more simple note taking experience.

As for the three points:
- I've been working on the syntax highlighting, however there are some basic limitations with the current parser/display engine that can't be overcome (things like tables will never be support with it).  Having said that, there is certainly room for improvement.
- Blocks do auto close in some cases, for example if you use the keyboard shortcuts (aka Ctrl-B for bold) instead of typing them, this could be expanded in the future.
- WebDav is way out of scope for Folio, the actual storage of the notes is up to the OS to handle.  However, having said that, you can certainly setup a system where the data directory is accessible through WebDav.  My setup is to store the data in the Nextcloud Notes directory, which then can expose it as a WebDav folder or sync it with other systems.

**Comment by somepaulo** [comment]
I came here just for the issue subject - code syntax highlighting, not the other two points. It would be amazing to have.
It would also be nice to have the code on a different background. Currently this is implemented with inline code between single ` marks, but not for blocks of code between ``` marks.
Thank you!

**Comment by toolstack** [comment]
@somepaulo I agree that code blocks should have the same background as code snippets, I'll likely add that to the next release.

Adding full syntax highlighting inside code blocks seems unlikely.
